core/src/main/java/de/unistuttgart/iaas/amyassist/amy/core/audio/environment/AbstractAudioEnvironment.java

```diff
@@ -0,0 +1,234 @@
+	 *            the data
+	 * @param off
+	 *            the start offset in the data
+	 * @param len
+	 *            the number of bytes to write
+	 */
+	protected abstract void writeToOutput(byte[] buffer, int off, int len);
+
+	/**
+	 * @see de.unistuttgart.iaas.amyassist.amy.core.audio.environment.AudioEnvironment#stopOutput()
+	 */
+	@Override
+	public void stopOutput() {
+		this.outputWorker.cancel(true);
+	}
+
+	/**
+	 * @see de.unistuttgart.iaas.amyassist.amy.core.audio.environment.AudioEnvironment#playAudio(de.unistuttgart.iaas.amyassist.amy.core.audio.AudioOutput,
+	 *      de.unistuttgart.iaas.amyassist.amy.core.audio.AudioManager.OutputBehavior)
+	 */
+	@Override
+	public void playAudio(AudioOutput audioToPlay, AudioManager.OutputBehavior behavior) {
+
+		if (!audioToPlay.getFormat().matches(getOutputFormat()))
+			throw new IllegalArgumentException("AudioToPlay has the wrong AudioFormat.");
+
+		synchronized (this.outputWorker.outputLock) {
+			switch (behavior) {
+			case INTERRUPT_ALL:
+				this.outputQueue.clear();
+				this.outputQueue.addFirst(audioToPlay);
+				this.outputWorker.cancel(true);
+				break;
+			case INTERRUPT_CURRENT:
+				this.outputQueue.addFirst(audioToPlay);
+				this.outputWorker.cancel(true);
+				break;
+			case QUEUE:
+				this.outputQueue.add(audioToPlay);
+				break;
+			case QUEUE_PRIORITY:
+				this.outputQueue.addFirst(audioToPlay);
+				break;
+			case SUSPEND:
+				this.outputQueue.addFirst(audioToPlay);
+				this.outputWorker.cancel(false, ao -> {
+					synchronized (this.outputWorker.outputLock) {
+						// This get's called before the above element get's taken out of the queue, so we need to take
+						// that and then re add it at the front of the queue.
+						AudioOutput newOutput = this.outputQueue.pollFirst();
+						this.outputQueue.addFirst(ao);
+						this.outputQueue.addFirst(newOutput);
+					}
+				});
+				break;
+			default:
+				throw new IllegalStateException("Unknown behavior");
+			}
+		}
+	}
+
+	/**
+	 * @see de.unistuttgart.iaas.amyassist.amy.core.audio.environment.AudioEnvironment#getAudioInputStream()
+	 */
+	@Override
+	public AudioInputStream getAudioInputStream() {
+		QueuedInputStream qis = new QueuedInputStream();
+		this.inputStreams.add(qis);
+		return new AudioInputStream(qis, getInputFormat(), AudioSystem.NOT_SPECIFIED);
+	}
+
+	/**
+	 * @see de.unistuttgart.iaas.amyassist.amy.core.audio.environment.AudioEnvironment#start()
+	 */
+	@Override
+	public void start() {
+		String aeId = this.getAudioEnvironmentIdentifier().toString();
+
+		this.outputThread = new Thread(this.outputWorker, "AE<" + aeId + ">OutputThread");
+		this.inputThread = new Thread(this.inputWorker, "AE<" + aeId + ">InputThread");
+
+		this.outputThread.start();
+		this.inputThread.start();
+	}
+
+	/**
+	 * @see de.unistuttgart.iaas.amyassist.amy.core.audio.environment.AudioEnvironment#stop()
+	 */
+	@Override
+	public void stop() {
+		this.outputThread.interrupt();
+		this.inputThread.interrupt();
```

Comment:
this does not work

Reply:
It does for me. Can you try again after the latest commit?